core/src/main/scala/akka/persistence/cassandra/journal/Buffer.scala

```diff
@@ -107,7 +107,7 @@ private[akka] case class Buffer(
   }
 
   final def addPending(write: AwaitingWrite): Buffer = {
-    copy(size = size + write.events.size, pending = pending :+ write)
+    copy(size = size + write.events.size, pending = pending :+ write, writeRequired = true)
```

Comment:
Seems innocent and correct but difficult to see the full consequences. Tests are failing.

I looked at other places where we add to `pending` and then it's indeed `writeRequired = true` aside from `rebuild`. Maybe it's always set to true in `rebuild` also via the `add` but that is not immediately obvious.